connman/src/service.c

```diff
@@ -9343,8 +9343,19 @@ bool __connman_service_create_from_network(struct connman_network *network)
 	if (__connman_network_get_weakness(network))
 		return true;
 
+	index = connman_network_get_index(network);
+
 	if (service->path) {
 		update_from_network(service, network);
+
```

Comment:
Is there some reason we can't do these if-sets right after getting the index, then we wouldn't need to have the same code twice?